e3db/sodium_crypto.py

```diff
@@ -76,6 +77,19 @@ def generate_keypair(self):
         key = nacl.public.PrivateKey.generate()
         return key.public_key, key
 
+    @classmethod
+    def generate_signing_key(self, key: bytes):
+        return nacl.signing.SigningKey(key)
+
+    @classmethod
+    def derive_public_signing_key(self, key: bytes):
+        pass
+
+    @classmethod
+    def sign_string(self, string_to_sign: bytes, key: bytes) -> nacl.signing.SignedMessage:
+        signing_key = self.generate_signing_key(key)
+        return signing_key.sign(string_to_sign)
```

Comment:
wondering if for end user comprehension we should add these methods to the nist_crypto class as well but have them raise a "Not Implemented exception"?

Reply:
Added this. Should I avoid adding the return type hint to its declaration to avoid another import (`nacl.Signing`) to nist_crypto? It might also match better with the other member methods of the class?